modules/stochastic_tools/src/samplers/AdaptiveImportanceSampler.C

```diff
@@ -0,0 +1,196 @@
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AdaptiveImportanceSampler.h"
+#include "Distribution.h"
+#include "Normal.h"
+#include "Uniform.h"
+
+registerMooseObjectAliased("StochasticToolsApp", AdaptiveImportanceSampler, "AdaptiveImportance");
+
+InputParameters
+AdaptiveImportanceSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Adaptive Importance Sampler.");
+  params.addRequiredParam<std::vector<DistributionName>>(
+      "distributions",
+      "The distribution names to be sampled, the number of distributions provided defines the "
+      "number of columns per matrix.");
+  params.addRequiredParam<ReporterName>("inputs_reporter", "Reporter with input parameters.");
+  params.addRequiredParam<std::vector<Real>>("proposal_std",
+                                             "Standard deviations of the proposal distributions");
+  params.addRequiredParam<Real>("output_limit", "Limiting values of the VPPs");
+  params.addRequiredParam<std::vector<Real>>(
+      "initial_values", "Initial input values to get the importance sampler started");
+  params.addRequiredParam<int>("num_samples_train",
+                               "Number of samples to learn the importance distribution");
+  params.addRequiredParam<Real>(
+      "std_factor", "Factor to be multiplied to the standard deviation of the importance samples");
+  params.addParam<bool>("use_absolute_value", false, "Use absolute value of the sub app output");
+  params.addParam<unsigned int>(
+      "num_random_seeds",
+      100000,
+      "Initialize a certain number of random seeds. Change from the default only if you have to.");
+  return params;
+}
+
+AdaptiveImportanceSampler::AdaptiveImportanceSampler(const InputParameters & parameters)
+  : Sampler(parameters),
+    ReporterInterface(this),
+    _distribution_names(getParam<std::vector<DistributionName>>("distributions")),
+    _proposal_std(getParam<std::vector<Real>>("proposal_std")),
+    _initial_values(getParam<std::vector<Real>>("initial_values")),
+    _output_limit(getParam<Real>("output_limit")),
+    _num_samples_train(getParam<int>("num_samples_train")),
+    _std_factor(getParam<Real>("std_factor")),
+    _use_absolute_value(getParam<bool>("use_absolute_value")),
+    _num_random_seeds(getParam<unsigned int>("num_random_seeds")),
+    _step(getCheckedPointerParam<FEProblemBase *>("_fe_problem_base")->timeStep()),
+    _perf_compute_sample(registerTimedSection("computeSample", 4))
+{
+  // Filling the `distributions` vector with the user-provided distributions.
+  for (const DistributionName & name : _distribution_names)
+    _distributions.push_back(&getDistributionByName(name));
```

Comment:
`_distribution_names` doesn't need to be a member variable if you do:

```suggestion
  for (const DistributionName & name : getParam<std::vector<DistributionName>>("distributions))
    _distributions.push_back(&getDistributionByName(name));
```